### ISSUE TITLE
Lots of ANR with com.instabug.library.b.c (b.java)?

### ISSUE THREAD

**Issue by raymondctc** [Instabug/Instabug-Android#209]
### Steps to Reproduce the Problem
We found that there are a number of ANRs, stacktrace is below
```
"main" prio=5 tid=1 Blocked
  | group="main" sCount=1 dsCount=0 flags=1 obj=0x76177ff8 self=0x7675014c00
  | sysTid=19615 nice=-10 cgrp=default sched=0/0 handle=0x76fb944560
  | state=S schedstat=( 107340554839 16256233335 203752 ) utm=7632 stm=3102 core=6 HZ=100
  | stack=0x7ffc231000-0x7ffc233000 stackSize=8MB
  | held mutexes=
 
  at com.instabug.library.b.c (b.java)
- waiting to lock <0x0d405e16> (a ixx) held by thread 259
 
  at com.instabug.library.network.a.onReceive (a.java:37)
 
  at android.app.LoadedApk$ReceiverDispatcher$Args.lambda$getRunnable$0 (LoadedApk.java:1483)
 
  at android.app.-$$Lambda$LoadedApk$ReceiverDispatcher$Args$_BumDX2UKsnxLVrE6UJsJZkotuA.run (lambda)
 
  at android.os.Handler.handleCallback (Handler.java:873)
 
  at android.os.Handler.dispatchMessage (Handler.java:99)
 
  at android.os.Looper.loop (Looper.java:214)
 
  at android.app.ActivityThread.main (ActivityThread.java:7050)
 
  at java.lang.reflect.Method.invoke (Method.java)
 
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:493)
 
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:965)
```

### Instabug integration code
```
new Instabug.Builder(application, application.getString(R.string.instabug_app_id))
                .setInvocationEvents(InstabugInvocationEvent.NONE)
                .build();
        Replies.setPushNotificationState(Feature.State.ENABLED);
        Instabug.setDebugEnabled(!AppBuildConfig.App.IS_RELEASE);
        Instabug.setPrimaryColor(ResourcesCompat.getColor(getResources(), R.color.under9_theme_black, null));
        // Arguments:initialScreenshot, extraScreenshot, galleryImage & ScreenRecording
        BugReporting.setAttachmentTypesEnabled(false, true, true, true);
        Surveys.setShouldShowWelcomeScreen(true);
        Chats.setState(Feature.State.DISABLED);
```

### SDK Version
8.5.0

### Android Version
All

### Device Model
N/A

**Comment by mzelzoghbi** [comment]
Hi @adavis @raymondctc, Thanks for reporting this, We created a **SNAPSHOT** that will fix the issue can you give it a try and tell us if it solved the issue?
```
implementation 'com.instabug.library:instabug:8.5.0.8-SNAPSHOT'
```
you will need to add these lines to your app **build.gradle** before your dependencies
```
repositories {
    ....
    maven {
        url "https://oss.sonatype.org/content/repositories/snapshots"
    }
}
```

**Comment by raymondctc** [comment]
@mzelzoghbi Thanks, it works 😄

**Comment by jimmy0251** [comment]
I'm still facing the same issue with sdk 8.6.0, is this supposed to be fixed? If not, when will this be released? I can't use 8.5.0 based snapshot as I will miss other important fixes of 8.6.0 version.

**Comment by amrbrkt** [comment]
Hi @jimmy0251, I've updated the snapshot with our recent version. 

Please uses `8.6.1.6-SNAPSHOT` and let me know if you still facing any problem. 


you will need to add these lines to your app build.gradle 

```
repositories {
    ....
    maven {
        url "https://oss.sonatype.org/content/repositories/snapshots"
    }
}
```

```
implementation 'com.instabug.library:instabug:8.5.0.8-SNAPSHOT'
```

**Comment by mzelzoghbi** [comment]
Hello everyone, We just released **8.6.2** that fixes the ANR issues. Please feel free to reopen this issue if you encounter the same issue again.

Thank You!
